lib/http/response/parser.rb

```diff
@@ -34,17 +34,12 @@ def on_headers_complete(headers)
       end
 
       def on_body(chunk)
-        if @chunk
-          @chunk << chunk
-        else
-          @chunk = chunk
-        end
+        @chunk ||= ''
```

Comment:
Yeah +1, this adds an additional allocation and makes the code more complicated, in addition to just breaking things.